### DISCUSSION TITLE
Specify app to be in Space, and order of the Spaces

### DISCUSSION THREAD

**Discussion by benhamilton** [ttscoff/bunch-docs#79] [Ideas]
There are some apps I often use in my "Work" bunch (Slack, Calendar, Outlook, etc) and I normally use those apps in full screen spaces. Would be great to be able to specify the order of those apps in those spaces.
i.e. Desktop 1, Slack, Calendar, Outlook
Where Slack is the space one to the right of Desktop 1, Calendar is two spaces to the right of Desktop 1 etc.

This would then make the four-finger left/right swipe through the spaces very second nature as I would just *know* that three four-finger swipes would take me to Outlook.

**Comment by ttscoff** [comment]
Sorry, there is no way to automate this. You can send keyboard shortcuts to make apps full screen, but Bunch can't control any of the ordering. Not even a dedicated window manager like Moom can control that.

- Brett

**Reply by basetta** [reply]
I am using 
```
# Switch to the first Desktop
* tell application "System Events" to key code 18 using control down
* delay 1
```
to switch to the first desktop for instance. And I am maximizing the windows. (Instead of full screen)

**Reply by benhamilton** [reply]
Cool. thanks for the pointer :)
The maximising rather than full screening is the key part there - spent a minute wondering why it wasn't working 🤣

**Reply by xtianmason** [reply]
@basetta could you please post an example? I’ma little slow on how this works but it would really help me.

**Reply by benhamilton** [reply]
I setup keyboard shortcuts to switch desktops in Mission Control as per screenshot here:
<img width="780" alt="CleanShot 2021-08-05 at 12 11 53@2x" src="https://user-images.githubusercontent.com/876530/128280026-be3b01bd-4cd5-433d-a16f-6453c6cd229e.png">

Switch to the desktop I'm wanting, open the app there, maximise the app (rather than full screen).
Now the apps are on individual desktops in the order I want them in.

**Reply by xtianmason** [reply]
> I setup keyboard shortcuts to switch desktops in Mission Control as per screenshot here:
> 
> <img width="780" alt="CleanShot 2021-08-05 at 12 11 53@2x" src="https://user-images.githubusercontent.com/876530/128280026-be3b01bd-4cd5-433d-a16f-6453c6cd229e.png">
> 
> 
> 
> Switch to the desktop I'm wanting, open the app there, maximise the app (rather than full screen).
> 
> Now the apps are on individual desktops in the order I want them in.

Now if we could put those shortcuts into bunch before the command to open that app, it would solve the problem. I think we are going to need Apple to develop better space control.

**Comment by SteveMattan** [comment]
I also use specific apps in specific spaces. What I've done is assigned each app to a specific desktop. The app will then open in that space. Of course, this will only work if you always want the app to be in that space.

I also open a set of folders in a specific space. I do this but first sending a global key combination, using the Mission Control keyboard shortcuts noted by others in this thread, to move to the desired space. Then I open the folders and tell Moom to arrange them via a snapshot (called via AppleScript). Then I open the apps using delays to ensure the folders are opened and Moomed.

What I've not been able to figure out how to do is to finish up back in that space with the open folders. I send the same global key combination as the last step in the bunch, using a delay, and the log shows it as the last item executed. But the system does not move to the specified space. Note that if I manually enter the key combination once the bunch completes it does move to the proper space.

Any suggestions on what to do here would be appreciated.

UPDATE: I figured it out. I ran the bunch with one of the applications already running. And the system transferred to the specified space. So I increased the delay time for the final key combination and voila it now works.

Stay well,

SteveM
